tasks/hyrax-metadata-updates/tests/test-index.js

```diff
@@ -0,0 +1,123 @@
+'use strict';
+
+const test = require('ava');
+const fs = require('fs');
+const xml2js = require('xml2js');
+const { promisify } = require('util');
+
+const xmlParseOptions = {
+  ignoreAttrs: true,
+  mergeAttrs: true,
+  explicitArray: false
+};
+
+const { InvalidArgument } = require('@cumulus/errors');
+const rewire = require('rewire');
+const HyraxMetadataUpdate = rewire('../index');
+
+const generateAddress = HyraxMetadataUpdate.__get__('generateAddress');
+const getGranuleUr = HyraxMetadataUpdate.__get__('getGranuleUr');
+const addHyraxUrl = HyraxMetadataUpdate.__get__('addHyraxUrl');
+
+test.afterEach.always(async () => {
+  delete process.env.CMR_ENVIRONMENT;
+});
+
+test('Test return prod OPeNDAP host when no environment value supplied', async (t) => {
+  delete process.env.CMR_ENVIRONMENT;
+  const actual = await generateAddress();
+  t.is(actual, 'https://opendap.earthdata.nasa.gov');
+});
+
+test('Test return prod OPeNDAP host when prod environment value supplied', async (t) => {
+  process.env.CMR_ENVIRONMENT = 'PROD';
+  const actual = await generateAddress();
+  t.is(actual, 'https://opendap.earthdata.nasa.gov');
+});
+
+test('Test return prod OPeNDAP host when ops environment value supplied', async (t) => {
+  process.env.CMR_ENVIRONMENT = 'OPS';
+  const actual = await generateAddress();
+  t.is(actual, 'https://opendap.earthdata.nasa.gov');
+});
+
+test('Test return sit OPeNDAP host when sit environment value supplied', async (t) => {
+  process.env.CMR_ENVIRONMENT = 'SIT';
+  const actual = await generateAddress();
+  t.is(actual, 'https://opendap.sit.earthdata.nasa.gov');
+});
+
+test('Test return uat OPeNDAP host when uat environment value supplied', async (t) => {
+  process.env.CMR_ENVIRONMENT = 'UAT';
+  const actual = await generateAddress();
+  t.is(actual, 'https://opendap.uat.earthdata.nasa.gov');
+});
+
+test('Test return error when invalid environment supplied for host generation', async (t) => {
+  process.env.CMR_ENVIRONMENT = 'FOO';
+  t.throws(
+    () => generateAddress(),
+    {
+      message: 'Environment foo is not a valid environment.',
+      instanceOf: InvalidArgument
+    }
+  );
+});
+
+test('Test granule ur extraction from UMM-G', async (t) => {
+  const data = fs.readFileSync('tests/data/umm-gin.json', 'utf8');
+  const metadata = JSON.parse(data);
+  const actual = await getGranuleUr(metadata, true);
```

Comment:
I believe this function does not need to be `async`, nor `await` as `getGranuleUr()` is synchronous.